service/src/main/kotlin/io/provenance/onboarding/frameworks/cee/P8eContractService.kt

```diff
@@ -26,26 +30,61 @@ class P8eContractService : ContractService {
         contractClass: Class<T>,
         records: Map<String, Message>,
         scopeUuid: UUID,
-        sessionUuid: UUID?
+        sessionUuid: UUID?,
+        participants: Map<Specifications.PartyType, Originator>?,
+        scope: ScopeResponse?,
     ): Session =
-        client
-            .newSession(contractClass, LoanScopeSpecification::class.java)
-            .setScopeUuid(scopeUuid)
-            .configureSession(records, sessionUuid)
-            .also { session ->
-                log.info("[L: ${session.scopeUuid}, S: ${session.sessionUuid}] ${contractClass.simpleName} has been setup.")
-            }
+        when (scope) {
+            null ->
+                client
+                    .newSession(contractClass, LoanScopeSpecification::class.java)
+                    .setScopeUuid(scopeUuid)
+                    .configureSession(records, sessionUuid, participants)
+                    .also { session ->
+                        log.info("[L: ${session.scopeUuid}, S: ${session.sessionUuid}] ${contractClass.simpleName} has been setup.")
+                    }
+            else ->
+                client
+                    .newSession(contractClass, scope)
+                    .configureSession(records, sessionUuid, participants)
```

Comment:
Mind clarifying for me why the session doesn't need to set the scope UUID if `scope` is not null?

Reply:
if the scope is found on prov, then the scope uuid is already populated. At that point, the only thing to do is to configure the session you want to act upon.